user_tools/src/spark_rapids_pytools/common/utilities.py

```diff
@@ -208,6 +209,26 @@ def gen_multiline_str(cls, *items) -> str:
     def get_os_name(cls) -> str:
         return os.uname().sysname
 
+    @classmethod
+    def get_value_or_pop(cls, provided_value, options_dict, short_flag, default_value=None):
+        """
+        Gets a value or pops it from the provided options dictionary if the value is not explicitly provided.
+
+        :param provided_value: The value to return if not None.
+        :param options_dict: Dictionary containing options.
+        :param short_flag: Flag to look for in options_dict.
+        :param default_value: The default value to return if the target_key is not found. Defaults to None.
+        :return: provided_value or the value from options_dict or the default_value.
+        """
+        if provided_value is not None:
+            return provided_value
+        if short_flag in options_dict:
+            if not cls.warning_issued:
+                cls.warning_issued = True
+                print('Warning: Instead of using short flags for argument, consider providing the value directly.')
```

Comment:
nit: it may be more clear to specify which short flag in the warning statement.

Reply:
I wanted to call it once for all the short flags that are passed. We can specify the short flag here but that would only correspond to the first one.

Reply:
Sorry I missed this. Thanks!